Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-oversight.controller.ts

```diff
@@ -56,13 +93,58 @@
+                .createUpdate("Bemærkninger", () => this.dataProcessingRegistrationService.updateOversightIntervalRemark(this.dataProcessingRegistration.id, oversightIntervalRemark))
                 .executeAsync(success => {
                     this.dataProcessingRegistration.oversightInterval.remark = oversightIntervalRemark;
                     this.bindOversightIntervalRemark();
                     return success;
                 });
         }
+
+        private changeIsOversightCompleted(isOversightCompleted: Models.ViewModel.Generic.Select2OptionViewModel<Models.Api.Shared.YesNoUndecidedOption>) {
+            this.apiUseCaseFactory
+                .createUpdate("Gennemført tilsyn", () => this.dataProcessingRegistrationService.updateOversightCompleted(this.dataProcessingRegistration.id, isOversightCompleted.optionalObjectContext))
+                .executeAsync(success => {
+                    if (success.optionalServerDataPush) {
+                        this.dataProcessingRegistration.oversightCompleted = success.optionalServerDataPush.oversightCompleted;
+                    }
+
+                    this.dataProcessingRegistration.oversightCompleted.value = isOversightCompleted.optionalObjectContext;
+                    this.bindOversightCompleted();
+                    this.bindLatestOversightCompletedDate();
+                    return success;
+                });
+
+        }
+
+        private changeLatestOversightCompletedDate(latestOversightCompletedDate: string) {
+            if (!!latestOversightCompletedDate) {
+                var formattedDate = Helpers.DateStringFormat.fromDDMMYYYYToYYYYMMDD(latestOversightCompletedDate);
+                if (!!formattedDate.convertedValue) {
+                    return this.apiUseCaseFactory
+                        .createUpdate("Dato for seneste tilsyn",
+                            () => this.dataProcessingRegistrationService.updateLatestOversightCompletedDate(
+                                this.dataProcessingRegistration.id,
+                                formattedDate.convertedValue))
+                        .executeAsync(success => {
+                            this.dataProcessingRegistration.oversightCompleted.optionalDateValue = latestOversightCompletedDate;
+                            this.bindLatestOversightCompletedDate();
+                            return success;
+                        });
+                }
+                return this.notify.addErrorMessage(formattedDate.errorMessage);
+            }
+        }
+
+        private changeOversightCompletedRemark(oversightCompletedRemark: string) {
+            this.apiUseCaseFactory
+                .createUpdate("Bemærkning", () => this.dataProcessingRegistrationService.updateOversightCompletedRemark(this.dataProcessingRegistration.id, oversightCompletedRemark))
```

Comment:
Mener også jeg har tilføjet en dataProcessingRegistrationId som kan bruges i stedet for at gå ind på objektet og hente id hver gang.